### ISSUE TITLE
deploy.mk file doesn't work in deploy procedure

### ISSUE THREAD

**Issue by salsaiu** [matrix-io/matrixio_hal_esp32#9]
I have bought your Matrix Voice esp32 device, after many configuration attempts (I followed https://www.hackster.io/matrix-labs/get-started-w-esp32-on-the-matrix-voice-d01e0d guide), I have loaded program, onto the ESP32, through a batch file created ad hoc. I believe that your bash code in deploy.mk has an error, because, doing the procedure from my PC, I obtain (in the step 3 "Build and Deploy") the following error:

esptool.py wrapper for MATRIX Voice
usage: esptool write_flash [-h] [–flash_freq {keep,40m,26m,20m,80m}]
[–flash_mode {keep,qio,qout,dio,dout}]
[–flash_size FLASH_SIZE]
[–spi-connection SPI_CONNECTION] [–no-progress]
[–verify] [–compress | --no-compress]
[ …]
esptool write_flash: error: argument : Must be pairs of an address and the binary filename to write there.

I have solved it, writing a batch file and launching it directly from /tmp directory of my Raspberry (instead of PC)
This is the batch file content, that help me to avoid the flash error:

voice_esptool --chip esp32 --port /dev/ttyS0 --baud 115200 --before default_reset --after hard_reset write_flash -u --flash_mode dio --flash_freq 40m --flash_size detect 0x1000 bootloader/bootloader.bin 0x10000 mic_energy.bin 0x8000 partitions_singleapp.bin

Could you fix deploy.mk file? Thanks
Regards
Salva

**Comment by tafaust** [comment]
I got the same problem here.

**Comment by azat0th** [comment]
Same problem here.

The make deploy produce a command like : 

voice_esptool --chip esp32 --port /dev/ttyS0 --baud 115200 --before default_reset --after hard_reset write_flash -u --flash_mode dio --flash_freq 40m --flash_size detect /tmp/build/ota_data_initial.bin 0x1000 /tmp/build/bootloader/bootloader.bin 0x10000 /tmp/build/mic_energy.bin 0x8000 /tmp/build/partitions_singleapp.bin

/tmp/build/ota_data_initial.bin doesn't exist in the /tmp/buid path of the raspberry, and there is no address for it in command line..

So the solution of salsaiu is working for me too.

**Comment by tafaust** [comment]
Yeah, it's working for me too but this is no real fix for any case, right?

**Comment by azat0th** [comment]
Finally, I've made a little fix directly in deploy.mk :

`DEPLOY_CMD = $(shell echo $(ESPTOOL_WRITE_FLASH) $(ESPTOOL_ALL_FLASH_ARGS) | sed -e "s=$(PWD)=/tmp=g")`
`DEPLOY_CMD_FIX = $(shell echo $(DEPLOY_CMD) | sed -e "s|/tmp/build/ota_data_initial.bin||")`
`tar cf - build/bootloader/bootloader.bin build/*.bin | ssh $(RPI_HOST) 'tar xf - -C /tmp;sudo voice_esp32_reset;$(DEPLOY_CMD_FIX)'`

This code just delete "/tmp/build/ota_data_initial.bin" from the deploy command. With this fix, the examples are working fine and it is easy to correct if needed.
But I don't know what is ota_data_initial.bin...
